packages/plugin/src/theia.proposed.textSearchQuery.d.ts

```diff
@@ -0,0 +1,225 @@
+// *****************************************************************************
+// Copyright (C) 2023 Ericsson and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+// code copied and modified from https://github.com/microsoft/vscode/blob/1.77.0/src/vscode-dts/vscode.proposed.textSearchQuery.d.ts
+
+export module '@theia/plugin' {
+
+    /**
+     * The parameters of a query for text search.
+     */
+    export interface TextSearchQuery {
+        /**
+         * The text pattern to search for.
+         */
+        pattern: string;
+
+        /**
+         * Whether or not `pattern` should match multiple lines of text.
+         */
+        isMultiline?: boolean;
+
+        /**
+         * Whether or not `pattern` should be interpreted as a regular expression.
+         */
+        isRegExp?: boolean;
+
+        /**
+         * Whether or not the search should be case-sensitive.
+         */
+        isCaseSensitive?: boolean;
+
+        /**
+         * Whether or not to search for whole word matches only.
+         */
+        isWordMatch?: boolean;
+    }
+
+    /**
+     * Options that can be set on a findTextInFiles search.
+     */
+    export interface FindTextInFilesOptions {
```

Comment:
This is a duplicate from theia.proposed.findTextInFiles.d.ts. This one may be removed?